bundler/lib/dependabot/bundler/update_checker/latest_version_finder.rb

```diff
@@ -44,21 +43,21 @@ def lowest_security_fix_version
                     :credentials, :ignored_versions, :security_advisories
 
         def fetch_latest_version_details
-          if dependency_source.is_a?(::Bundler::Source::Git)
-            return latest_git_version_details
+          if dependency_source.git?
+            return dependency_source.latest_git_version_details
```

Comment:
This is a little bit of a smell, I couldn't find a better way to abstract this this method either directly returns a has obtained from the dependency file via Bundler, or a hash it creates via filtering an array of `Gem::Version` objects.

This seemed like the least obtrusive way to handle it.

Reply:
Could we also call `dependency_source.versions` here? It seems like that uses the same `#git?` check to call this method? That way we may be able to hide some more of the implementation?

Reply:
Ah, I think that might be a hangover I meant to remove - passing back an array of one hash vs an array of gems means this method would have to check the contents of what it was given which is feels kind of gnarly.

I did briefly think about doing a double return of `versions, commit` to clarify the test in this class, but that seemed like a weird idiom.

Reply:
I elected to stick with `#versions` returning an empty array for git to mirror `#latest_git_version_details` returning nil for !git. Adding some yard docs and comments help explain it better than a weird multiple return, etc.